core/docz-utils/src/jsx.ts

```diff
@@ -2,19 +2,29 @@ import * as jsxUtils from 'jsx-ast-utils'
 import strip from 'strip-indent'
 import escapeJS from 'js-string-escape'
 
-import { valueFromTraverse, codeFromNode } from './ast'
+import { valueFromTraverse } from './ast'
 
 export const propFromElement = (prop: string) =>
   valueFromTraverse(
     p => p.isJSXOpeningElement(),
     p => jsxUtils.getPropValue(jsxUtils.getProp(p.node.attributes, prop))
   )
 
-export const removeTags = (code: string) => {
-  const open = codeFromNode(p => p.isJSXOpeningElement())
-  const close = codeFromNode(p => p.isJSXClosingElement())
+const getTagContentsRange = valueFromTraverse<[number, number] | null>(
+  p => p.isJSXElement(),
+  ({ node }) => {
+    if (!node.closingElement) {
+      // if the JSX element doesn't have a closingElement, it's because it's self-closed
+      // and thus does not have any content: <Playground />
+      return null
```

Comment:
what about returning `[0, 0]` as the default value?

Reply:
With invalid inputs (i.e. if it can't find a JSX element) the function returned by `valueFromTraverse` returns an empty string, so we still need the default value fallback when calling `getTagContentsRange` on line 26.
- I don't think this will happen in practice, but it's better to be safe and avoid crashing
- Otherwise Typescript complaints

Since we need the default value there anyway, returning something "falsy" here allows us to specify the default value only once.

Reply:
interesting to know that, it's fair enough! well done btw